tests/test_contact_specific_functions.py

```diff
@@ -315,6 +441,7 @@ def test_calculate_contact_forces_self_rod():
 
     "Test values"
     """Resulting forces act along the y-axis only.
+    Since the rod is stationary i.e velocity = 0, the net force is a function of contact force only.
     The net force was calculated by halving the contact force i.e
                                         net force = 0.5 * contact force = -0.5;
                                             where, contact force = k(1) * minimum distance between colliding elements centres(-1) * gamma(1) = -1
```

Comment:
Since your shape is symmetric the external forces from self contact should also be symmetric/opposite and equal, I think they are not now because the tangents and lengths need correction

Reply:
Okay, I shall reword this function again.